engine/src/main/scala/cromwell/engine/workflow/SingleWorkflowRunnerActor.scala

```diff
@@ -118,27 +123,49 @@ class SingleWorkflowRunnerActor(source: WorkflowSourceFiles, metadataOutputPath:
     ()
   }
 
-  private def requestStatus(): Unit = {
+  private def requestStatus(id: WorkflowId): Unit = {
     // This requests status via the metadata service rather than instituting an FSM watch on the underlying workflow actor.
     // Cromwell's eventual consistency means it isn't safe to use an FSM transition to a terminal state as the signal for
     // when outputs or metadata have stabilized.
-    val metadataBuilder = context.actorOf(MetadataBuilderActor.props(serviceRegistryActor), s"StatusRequest-Workflow-${stateData.id.get}-request-${UUID.randomUUID()}")
-    metadataBuilder ! GetStatus(stateData.id.get)
+    val metadataBuilder = context.actorOf(MetadataBuilderActor.props(serviceRegistryActor), s"StatusRequest-Workflow-$id-request-${UUID.randomUUID()}")
+    metadataBuilder ! GetStatus(id)
   }
 
-  private def issueReply(data: RunnerData): State = {
-    data.terminalState foreach { state => log.info(s"$Tag workflow finished with status '$state'.") }
-    data.failures foreach { e => log.error(e, e.getMessage) }
+  private def issueSuccessReply(replyTo: ActorRef): State = {
+    replyTo.tell(msg = (), sender = self) // Because replyTo ! () is the parameterless call replyTo.!()
+    context.stop(self)
+    stay()
+  }
 
-    val message: Any = data.terminalState collect { case WorkflowSucceeded => () } getOrElse Status.Failure(data.failures.head)
-    data.replyTo foreach  { _ ! message }
+  private def issueFailureReply(replyTo: ActorRef, e: Throwable): State = {
+    replyTo ! Status.Failure(e)
     context.stop(self)
-    stay
+    stay()
+  }
+
+  private def issueReply(data: CompletedSwraData) = {
+    data match {
+      case s: SucceededSwraData => issueSuccessReply(s.replyTo)
+      case f: FailedSwraData => issueFailureReply(f.replyTo, f.failure)
+      case a: AbortedSwraData => issueSuccessReply(a.replyTo)
+
+    }
   }
 
-  private def failAndFinish(e: Throwable): State = {
+  private def failAndFinish(e: Throwable, data: SwraData): State = {
     log.error(e, s"$Tag received Failure message: ${e.getMessage}")
-    issueReply(stateData.addFailure(e))
+    data match {
+      case EmptySwraData =>
+        log.error(e, s"Cannot issue response for {}. Need a 'replyTo' address to issue the response!", e.getMessage)
```

Comment:
also isn't interpolating the `e.getMessage` into the message going to be redundant with what happens passing the exception as the first parameter?